### PR TITLE
[RHPAM-2934] Performance bottleneck search variables in Postgres

### PR DESCRIPTION
**Pull request by elguardian**

### CODE REVIEW

**Comment by elguardian** [comment]
depends on https://github.com/kiegroup/jbpm/pull/1641

**Comment by gmunozfe** [comment]
In mysql drop scripts also remove these lines:

https://github.com/kiegroup/jbpm/blob/663505d8ff05fbb196c5a2684624b081b5c306f8/jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysql5/mysql5-jbpm-drop-schema.sql#L32

https://github.com/kiegroup/jbpm/blob/663505d8ff05fbb196c5a2684624b081b5c306f8/jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysqlinnodb/mysql-innodb-jbpm-drop-schema.sql#L32

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysql5/mysql5-jbpm-schema.sql [hunk 1]
```diff
@@ -689,17 +689,13 @@
...
+        add index IDX_PAsPot_TaskEntity (task_id, entity_id), 
         add constraint FK_tee3ftir7xs6eo3fdvi3xw026 
         foreign key (entity_id) 
         references OrganizationalEntity (id);
 
-    alter table PeopleAssignments_PotOwners 
-        add index IDX_PAsPot_Task (task_id), 
-        add constraint FK_4dv2oji7pr35ru0w45trix02x 
-        foreign key (task_id) 
-        references Task (id);
```
Comment (gmunozfe) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysql5/mysql5-jbpm-schema.sql:702]: Why removing foreign key and constraint?

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysqlinnodb/mysql-innodb-jbpm-schema.sql [hunk 2]
```diff
@@ -692,18 +692,13 @@
...
+        add index IDX_PAsPot_TaskEntity (task_id, entity_id), 
         add constraint FK_tee3ftir7xs6eo3fdvi3xw026 
         foreign key (entity_id) 
         references OrganizationalEntity (id);
 
-    alter table PeopleAssignments_PotOwners 
-        add index IDX_PAsPot_Task (task_id), 
-        add constraint FK_4dv2oji7pr35ru0w45trix02x 
-        foreign key (task_id) 
-        references Task (id);
```
Comment (gmunozfe) [inline jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysqlinnodb/mysql-innodb-jbpm-schema.sql:705]: Same here, why removing this?

**Comment by lazarotti** [comment]
@elguardian @mareknovotny  I was comparing the files that we have in product add-ons zip, inside the folder rhpam-7.7.0-add-ons/rhpam-7.7.0-migration-tool/upgrade-scripts/<database> - there we only have the files "bpms-*", jbpm-*" are not there, and it only contains product versions. So while "upgrade-scripts" use upstream version like jbpm-7.37-to-7.38.sql we also need to have as part of this  commit bpms-7.7-to-7.8.sql. I see that you created the file bpms-7.37-to-7.38.sql, you should change it to the product versions. Marek, i don't know how productization only get the bpms-* files, but this is how i see in past releases.

**Comment by lazarotti** [comment]
just as a note, compare the scripts "bpms-7.0-to-7.1.sql" and "jbpm-7.7-to-7.11.sql", they were the same situation (product/project)

**Comment by mareknovotny** [comment]
the assembly is filtering files from unpacked jar artifact folder see https://github.com/jboss-integration/rhba/blob/master/rhpam/src/main/assembly/migration-tool.zip.xml#L68-L91

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysqlinnodb/mysql-innodb-jbpm-schema.sql (hunk 2 above)
Comment (elguardian) [reply jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysqlinnodb/mysql-innodb-jbpm-schema.sql:705]: ok corrected

File: jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysql5/mysql5-jbpm-schema.sql (hunk 1 above)
Comment (elguardian) [reply jbpm-db-scripts/src/main/resources/db/ddl-scripts/mysql5/mysql5-jbpm-schema.sql:702]: corrected

**Comment by elguardian** [comment]
> @elguardian @mareknovotny I was comparing the files that we have in product add-ons zip, inside the folder rhpam-7.7.0-add-ons/rhpam-7.7.0-migration-tool/upgrade-scripts/ - there we only have the files "bpms-_", jbpm-_" are not there, and it only contains product versions. So while "upgrade-scripts" use upstream version like jbpm-7.37-to-7.38.sql we also need to have as part of this commit bpms-7.7-to-7.8.sql. I see that you created the file bpms-7.37-to-7.38.sql, you should change the name to the product versions in that case. Marek, i don't know how productization only get the bpms-* files, but this is how i see in past releases.

@lazarotti @mareknovotny 
cannot do that because there are scripts from 7.7 to 7.11. So this is out of question.

**Comment by elguardian** [comment]
> Use case-sensitive for dropping indexes in the new upgrade scripts:
> DROP INDEX IDX_PAsPot_Entity
> DROP INDEX IDX_PAsPot_Task
> I think failures in upgrade tests are related to this

funny thing this indexes wheren't there in first place.... only in the ddl but never happen in upgrades. so there is no need to drop them.

**Comment by mareknovotny** [comment]
> > @elguardian @mareknovotny I was comparing the files that we have in product add-ons zip, inside the folder rhpam-7.7.0-add-ons/rhpam-7.7.0-migration-tool/upgrade-scripts/ - there we only have the files "bpms-_", jbpm-_" are not there, and it only contains product versions. So while "upgrade-scripts" use upstream version like jbpm-7.37-to-7.38.sql we also need to have as part of this commit bpms-7.7-to-7.8.sql. I see that you created the file bpms-7.37-to-7.38.sql, you should change the name to the product versions in that case. Marek, i don't know how productization only get the bpms-* files, but this is how i see in past releases.
> 
> @lazarotti @mareknovotny
> cannot do that because there are scripts from 7.7 to 7.11. So this is out of question.

@elguardian where do you see them?  @lazarotti talked only about **bpms-*.sql** files
see my listing
```
 (master) find jbpm-db-scripts/src/main/resources/db/upgrade-scripts -type f -name bpms*.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/derby/bpms-7.0-to-7.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/derby/bpms-6.0-to-6.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/derby/bpms-6.4-to-7.0.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/derby/bpms-6.2-to-6.3.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/derby/bpms-6.1-to-6.2.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/derby/bpms-6.3-to-6.4.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/hsqldb/bpms-7.0-to-7.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/hsqldb/bpms-6.0-to-6.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/hsqldb/bpms-6.4-to-7.0.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/hsqldb/bpms-6.2-to-6.3.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/hsqldb/bpms-6.1-to-6.2.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/hsqldb/bpms-6.3-to-6.4.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/db2/bpms-7.0-to-7.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/db2/bpms-6.0-to-6.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/db2/bpms-6.4-to-7.0.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/db2/bpms-6.2-to-6.3.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/db2/bpms-6.1-to-6.2.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/db2/bpms-6.3-to-6.4.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysqlinnodb/bpms-7.0-to-7.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysqlinnodb/bpms-6.0-to-6.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysqlinnodb/bpms-6.4-to-7.0.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysqlinnodb/bpms-6.2-to-6.3.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysqlinnodb/bpms-6.1-to-6.2.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysqlinnodb/bpms-6.3-to-6.4.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysql5/bpms-7.0-to-7.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysql5/bpms-6.0-to-6.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysql5/bpms-6.4-to-7.0.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysql5/bpms-6.2-to-6.3.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysql5/bpms-6.1-to-6.2.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/mysql5/bpms-6.3-to-6.4.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/oracle/bpms-7.0-to-7.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/oracle/bpms-6.0-to-6.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/oracle/bpms-6.4-to-7.0.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/oracle/bpms-6.2-to-6.3.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/oracle/bpms-6.1-to-6.2.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/oracle/bpms-6.3-to-6.4.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sybase/bpms-7.0-to-7.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sybase/bpms-6.0-to-6.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sybase/bpms-6.4-to-7.0.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sybase/bpms-6.2-to-6.3.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sybase/bpms-6.1-to-6.2.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sybase/bpms-6.3-to-6.4.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/h2/bpms-7.0-to-7.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/h2/bpms-6.0-to-6.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/h2/bpms-6.4-to-7.0.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/h2/bpms-6.2-to-6.3.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/h2/bpms-6.1-to-6.2.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/h2/bpms-6.3-to-6.4.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver/bpms-7.0-to-7.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver/bpms-6.0-to-6.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver/bpms-6.4-to-7.0.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver/bpms-6.2-to-6.3.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver/bpms-6.1-to-6.2.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver/bpms-6.3-to-6.4.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver2008/bpms-7.0-to-7.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver2008/bpms-6.0-to-6.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver2008/bpms-6.4-to-7.0.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver2008/bpms-6.2-to-6.3.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver2008/bpms-6.1-to-6.2.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/sqlserver2008/bpms-6.3-to-6.4.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/postgresql/bpms-7.0-to-7.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/postgresql/bpms-6.0-to-6.1.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/postgresql/bpms-6.4-to-7.0.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/postgresql/bpms-6.2-to-6.3.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/postgresql/bpms-6.1-to-6.2.sql
jbpm-db-scripts/src/main/resources/db/upgrade-scripts/postgresql/bpms-6.3-to-6.4.sql
```

**Comment by elguardian** [comment]
@mareknovotny changed.

**Pull request merged**